scripts/unit_test/rootless_container.py

```diff
@@ -0,0 +1,83 @@
+"""rootless_container.py: Create a rootless container on any Linux and GitHub CI"""
+import ctypes
+import os
+
+# Unshare the user namespace, so that the calling process is moved into a new
+# user namespace which is not shared with any previously existing process.
+# Needed so that the current user id can be mapped to 0 for getting a new
+# mount namespace.
+CLONE_NEWUSER = 0x10000000
+# Unshare the mount namespace, so that the calling process has a private copy
+# of its root directory namespace which is not shared with any other process:
+CLONE_NEWNS = 0x00020000
+# Flags for mount(2):
+MS_BIND = 4096
+MS_REC = 16384
+MS_PRIVATE = 1 << 18
+
+
+def unshare(flags):  # type:(int) -> None
+    """Wrapper for the library call to unshare Linux kernel namespaces"""
+    lib = ctypes.CDLL(None, use_errno=True)
+    lib.unshare.argtypes = [ctypes.c_int]
+    rc = lib.unshare(flags)
+    if rc != 0:  # pragma: no cover
+        errno = ctypes.get_errno()
+        raise OSError(errno, os.strerror(errno), flags)
+
+
+def mount(source="none", target="", fs="", flags=0, options=""):
+    # type:(str, str, str, int, str) -> None
+    """Wrapper for the library call mount(). Supports Python2.7 and Python3.x"""
+    lib = ctypes.CDLL(None, use_errno=True)
+    lib.mount.argtypes = (
+        ctypes.c_char_p,
+        ctypes.c_char_p,
+        ctypes.c_char_p,
+        ctypes.c_ulong,
+        ctypes.c_char_p,
+    )
+    result = lib.mount(
+        source.encode(), target.encode(), fs.encode(), flags, options.encode()
+    )
+    if result < 0:  # pragma: no cover
+        errno = ctypes.get_errno()
+        raise OSError(
+            errno,
+            "mount " + target + " (" + options + "): " + os.strerror(errno),
+        )
+
+
+def umount(target):  # type:(str) -> None
```

Comment:
this function is not called, did you intention to call it automatically when leave the `session`?